hoot-core/src/main/cpp/hoot/core/io/HootApiDb.h

```diff
@@ -278,6 +279,29 @@ class HootApiDb : public ApiDb
 
   virtual long getNextId(const ElementType& elementType);
 
+  /**
+   * Updates an existing job status; assumes the job record already exists
+   *
+   * @param jobId ID of an existing job
+   * @param status status text to update the job status field with
+   */
+  void updateJobStatus(const QString jobId, const QString status);
+
+  /**
+   * Determines whether a job status exists
+   *
+   * @param jobId the ID of the job status to determine the existence of
+   * @return true if it exists; false otherwise
+   */
+  bool jobStatusExists(const QString jobId);
+
+  /**
+   * Inserts a new job status
+   *
+   * @param jobId the ID of the newly inserted job status
+   */
+  void insertJobStatus(const QString jobId);
+
 protected:
```

Comment:
Prior to these changes it was only services updating job status, but for this change core needed to do it.  I kind of hacked the job status to store the output changeset file names...open to better suggestions, that was the best idea I had so far.